source/material_model/steinberger.cc

```diff
@@ -461,6 +461,46 @@ namespace aspect
       return true;
     }
 
+
+
+    template <int dim>
+    std_cxx1x::array<std::pair<double, unsigned int>,2>
+    Steinberger<dim>::
+    enthalpy_derivative (const typename Interface<dim>::MaterialModelInputs &in) const
+    {
+      // We have to take into account here that the p,T spacing of the table of material properties
+      // we use might be on a finer grid than our model. Because of that we compute the enthalpy
+      // derivatives by using finite differences that average over the whole temperature and
+      // pressure range that is used in this cell. This way we should not miss any phase transformation.
+      std_cxx1x::array<std::pair<double, unsigned int>,2> derivative;
+
+      // get the pressures and temperatures at the vertices of the cell
+      const QTrapez<dim> quadrature_formula;
+      const unsigned int n_q_points = quadrature_formula.size();
+      FEValues<dim> fe_values (this->get_mapping(),
+                               this->get_fe(),
+                               quadrature_formula,
+                               update_values);
+
+      std::vector<double> temperatures(n_q_points), pressures(n_q_points);
+      fe_values.reinit (in.current_cell);
+
+      fe_values[this->introspection().extractors.temperature]
+      .get_function_values (this->get_current_linearization_point(), temperatures);
+      fe_values[this->introspection().extractors.pressure]
+      .get_function_values (this->get_current_linearization_point(), pressures);
+
+      // compute the averaged enthalpy derivatives for all temperatures and
+      // pressures in this cell
+      derivative = material_lookup[0]->enthalpy_derivatives(temperatures,
+                                                            pressures,
+                                                            1);
```

Comment:
Can you document somewhere that the 1 means you do one substep?

Reply:
And it would be great if you could add what the second output of that function is (the integers), because I don't think that's documented anywhere (so you might want to add that in the grain size model, and in this file just add a short note below where it is used).